server/@types/models/CourseParticipation.ts

```diff
@@ -25,8 +25,22 @@ type CourseParticipation = {
   source?: string
 }
 
+type CourseParticipationUpdate = {
+  outcome: CourseParticipationOutcome
+  setting: CourseParticipationSetting
+  source: string
```

Comment:
This could probably be `CourseParticipation['source']` too, in the spirit of the following two lines

Reply:
Good spot